core/src/main/java/com/microsoft/applicationinsights/internal/agent/CoreAgentNotificationsHandler.java

```diff
@@ -163,6 +163,30 @@ public void httpMethodFinished(String identifier, String method, String correlat
         telemetryClient.track(telemetry);
     }
 
+    @Override
+    public void httpMethodFinishedWithPath(String identifier, String method, String path, String correlationId, String uri, String target, int result, long delta) {
+        if (!LocalStringsUtils.isNullOrEmpty(uri) && (uri.startsWith("https://dc.services.visualstudio.com") || uri.startsWith("https://rt.services.visualstudio.com"))) {
+            return;
+        }
+        long deltaInMS = nanoToMilliseconds(delta);
+        String name = method + " " + path;
+        RemoteDependencyTelemetry telemetry = new RemoteDependencyTelemetry(name, uri, new Duration(delta), true);
```

Comment:
what about start time? It looks like now dependency would start Now and finish in future. It would affect trace viewer and provide inaccurate experience.

Reply:
About this one I am think numbers won't mess up. The logic to calculate the difference is in the Instrumented code. We calculare the difference there and the Remote Dependency Telemetry Object ultimately stores the execution time taken by dependency. See my other comment.

Reply:
Well, there is no code that sets telemetry.timestamp in this method. If it's not set here, it will be set in TelemetryClient.track to Now
```java
        if (telemetry.getTimestamp() == null) {
            telemetry.setTimestamp(new Date());
        }
```

i.e. duration is correct, but start time is not. I have not tried it though, it would be great if you can confirm it's not the case.